config/initializers/_fake_auth_configuration.rb

```diff
@@ -0,0 +1,5 @@
+# This configuration needs to load before the devise initializer so this value
+# is available when needed.
+
+require "#{Rails.root}/app/models/concerns/fake_auth_config.rb"
+Rails.application.config.fake_auth_enabled = FakeAuthConfig.fake_auth_status
```

Comment:
I'm somewhat curious why we used `Rails.application.config.fake_auth_enabled` instead of just calling `FakeAuthConfig.fake_auth_status` throughout. I know this is an ETD pattern carried over, but I'm not currently understanding the added value fwiw.

Reply:
It does feel DRYer to me to do it in an initializer. That way, we don't have to include the module everywhere we use it. That's only in three places currently, so ymmv as to how much value it adds in practice.

One thing I did consider while copying this logic over, and am considering more seriously now, is whether we really need the `FakeAuthConfig` module at all. What's the real value of using that over, say, a Flipflop feature? We basically would just need to add `ENV['FAKE_AUTH_CONFIG']` (or `ENV['FAKE_AUTH_ENABLED']` as it will so be named) in the pipeline and in our local dev environments, right? That doesn't feel any different from how we handle, e.g., API endpoints across multiple environments.

Reply:
Following some additional discussion in Slack, I think I prefer using the initializer and calling `Rails.application.config.fake_auth_enabled` rather than `FakeAuthConfig` for consistency's sake.

Reply:
Further amending my take on this. 🫠 I'm dropping the initializer due to the filenaming kludge required (i.e., it must load before the devise initializer so the config will be available there). It feels less wonky to me to require the FakeAuthConfig module where we need it.